code/modules/clothing/suits/marine_armor.dm

```diff
@@ -527,6 +527,36 @@
 
 	updateUsrDialog()
 
+/obj/item/clothing/suit/storage/marine/specialist/melee
+	name = "\improper B18-P offensive armor"
+	desc = "A modified prototype variant of the B18 defensive armor. Most of the protective layers of armor are replaced with smart prototype hydralics that assist with movement and agility to help you avoid those pesky acid splashes.\nHas an automated diagnostics and medical system for keeping its wearer alive."
+	icon_state = "melee_armor"
+	armor = list("melee" = 65, "bullet" = 40, "laser" = 40, "energy" = 25, "bomb" = 30, "bio" = 10, "rad" = 10, "fire" = 30, "acid" = 50)
+	flags_item_map_variant = NONE
+	slowdown = SLOWDOWN_ARMOR_LIGHT
+	var/stored_skill = -99 //To be safe.
+
+/obj/item/clothing/suit/storage/marine/specialist/melee/equipped(mob/user, slot)
+
+	. = ..()
+
+	if(user && user.mind && user.mind.cm_skills && slot == SLOT_WEAR_SUIT)
+		stored_skill = user.mind.cm_skills.firearms
+		user.mind.cm_skills.firearms = 0
+		to_chat(user,"<span class='warning'>Your bulky hands and armor make it difficult to use weapons...</span>")
+
+	return .
+
+
+/obj/item/clothing/suit/storage/marine/specialist/melee/dropped(mob/user)
+
+	. = ..()
+
+	if(user && user.mind && user.mind.cm_skills && stored_skill != -99)
```

Comment:
Use a define instead of `-99` for clarity.
You'll want to reset the value afterwards.